pymeasure/instruments/sorensen/sorensen30035E.py

```diff
@@ -0,0 +1,173 @@
+        in Volts, which can take floating point values between 300 and 300 V. """,
+        validator=strict_range,
+        values=[-300, 300]
+    )
+
+    voltage_limit = Instrument.control(
+        "SOUR:VOLT:LIM?", "SOUR:VOLT:LIM %g",
+        """ Sets an upper soft limit on the programmed output voltage
+        for the supply. Can take values floating point from 0 and 300 V. """,
+        validator=strict_range,
+        values=[0, 300]
+    )
+
+    @property
+    def output_enabled(self):
+        mapper = {'0': False, '1': True}
+        return mapper[self.ask("OUTPut:ISOLation?")]
+
+    @output_enabled.setter
+    def output_enabled(self, state):
+        if state:
+            if not self.output_enabled:
+                self.write("OUTPut:ISOLation 1")
+        elif not state:
+            if self.output_enabled:
+                if self.current_setpoint > 0:
+                    self.ramp_to_current(0, int(self.current_setpoint) * 20)
+                    while self.current > 0.01:
+                        time.sleep(.1)
+                    self.write("OUTPut:ISOLation 0")
+
+    def ramp_to_voltage(self, value, time=10):
+        """ Utilizes built in ramp function of the power supply to ramp voltage
+         to the specified value in the requested time. The large capacitance used to
+         suppress ripple in the power supply may affect charging/discharging times for
+         low currents/high impedance samples. """
+        if value > 300:
+            raise ValueError("Requested voltage too large, |V| must be 300 V or less")
+        self.write("SOUR:VOLT:RAMP %g %g" % (value, time))
+
+    def ramp_to_current(self, value, time=10):
+        """ Utilizes built in ramp function of the power supply to ramp voltage
+         to the specified value in the requested time. The large capacitance used to
+         suppress ripple in the power supply may affect charging/discharging times for
+         low currents/high impedance samples. """
+        if value > 3.5:
+            raise ValueError("Requested current too large, |I| must be 3.5 A or less")
+        self.write("SOUR:CURR:RAMP %g %g" % (value, time))
+
+    def abort_ramp(self):
+        """ Aborts all in-progress ramps
+        """
+        self.write("SOUR:CURR:RAMP:ABOR")
+        self.write("SOUR:VOLT:RAMP:ABOR")
+
+    def shutdown(self):
+        """ Disables the  output after ramping down at 1 V/s if voltage is non-zero."""
+        voltage = self.voltage
+        current = self.current
+        if voltage > 0.1 or current > 0.008:
+            self.ramp_to_voltage(0, voltage)
+            i = 0
+        while self.voltage > 0.1:
+            time.sleep(1)
+            i = i + 1
+            if i > 3 * voltage:
+                raise ValueError("Voltage has not reached zero in reasonable time,"
+                                 "giving up, OUTPUT IS STILL LIVE")
+        self.output_enabled = False
```

Comment:
Please call the super shutdown (with logging etc.):
```suggestion
        self.output_enabled = False
        super().shutdown()
```